controllers/csiaddonsnode_controller.go

```diff
@@ -39,17 +49,85 @@ type CSIAddonsNodeReconciler struct {
 
 // Reconcile is part of the main kubernetes reconciliation loop which aims to
 // move the current state of the cluster closer to the desired state.
-// TODO(user): Modify the Reconcile function to compare the state specified by
-// the CSIAddonsNode object against the actual cluster state, and then
-// perform operations to make the cluster state reflect the state specified by
-// the user.
-//
-// For more details, check Reconcile and its Result here:
-// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.10.0/pkg/reconcile
 func (r *CSIAddonsNodeReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
+	logger := log.FromContext(ctx)
+
+	// Fetch CSIAddonsNode instance
+	instance := &csiaddonsv1alpha1.CSIAddonsNode{}
+	err := r.Client.Get(ctx, req.NamespacedName, instance)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			logger.Info("CSIAddonsNode resource not found")
+
+			return ctrl.Result{}, nil
+		}
+
+		return ctrl.Result{}, err
+	}
+
+	nodeID := instance.Spec.Driver.NodeID
+	driverName := instance.Spec.Driver.Name
+	endPoint := instance.Spec.Driver.EndPoint
+	key := r.createKey(instance.Namespace, instance.Name)
+
+	logger = logger.WithValues("NodeID", nodeID, "EndPoint", endPoint, "Driver Name", driverName)
+
+	if !instance.DeletionTimestamp.IsZero() {
+		logger.Info("Deleting connection")
+		r.ConnPool.Delete(key)
+
+		if util.ContainsInSlice(instance.Finalizers, csiAddonsNodeFinalizer) {
+			logger.Info("Removing finalizer")
+			instance.Finalizers = util.RemoveFromSlice(instance.Finalizers, csiAddonsNodeFinalizer)
+			if err = r.Client.Update(ctx, instance); err != nil {
+				logger.Error(err, "Failed to remove finalizer")
+
+				return ctrl.Result{}, err
+			}
+		}
 
-	// TODO(user): your logic here
+		return ctrl.Result{}, nil
+	}
+
+	if !util.ContainsInSlice(instance.Finalizers, csiAddonsNodeFinalizer) {
+		logger.Info("Adding finalizer")
+
+		instance.Finalizers = append(instance.Finalizers, csiAddonsNodeFinalizer)
+		if err = r.Client.Update(ctx, instance); err != nil {
```

Comment:
I think using `Patch()` is better for most cases?

no need to place the assignment and check on the same line, split it like done elsewhere and stick to the same coding convention everywhere

Reply:
> I think using `Patch()` is better for most cases?

The examples I have seen use Update
https://github.com/csi-addons/volume-replication-operator/blob/05f25d5bffac00533e6acfa12709991d43336337/controllers/finalizers.go#L40
https://github.com/red-hat-storage/odf-operator/blob/0d5dd3ad5248c1dc3df62a9425bf375bddc77774/controllers/storagesystem_controller.go#L126
https://github.com/red-hat-storage/ocs-operator/blob/7438d205e74a63de1ccf52c2902d38923c5e3f62/controllers/storagecluster/reconcile.go#L309

> no need to place the assignment and check on the same line, split it like done elsewhere and stick to the same coding convention everywhere

I believe it is cleaner to user shorthand when only error is being returned, looks cleaner easier to read for me :sweat_smile:

Reply:
I do not think it is easier to read. The check `err != nil` (or something else) is less obvious as it is moved completely to the far right if the line.